crates/holochain/src/core/workflow/publish_dht_ops_workflow/unit_tests.rs

```diff
@@ -262,6 +264,108 @@ async fn ignores_data_by_other_authors() {
+        }),
+    };
+    let action = Action::Create(create_action.clone());
+    let new_entry_action = NewEntryAction::Create(create_action.clone());
+
+    let register_agent_activity_op = DhtOpHashed::from_content_sync(
+        ChainOp::RegisterAgentActivity(fixt!(Signature), action.clone()),
+    );
+    let store_entry_op = DhtOpHashed::from_content_sync(ChainOp::StoreEntry(
+        fixt!(Signature),
+        new_entry_action.clone(),
+        fixt!(Entry),
+    ));
+    let store_record_op = DhtOpHashed::from_content_sync(ChainOp::StoreRecord(
+        fixt!(Signature),
+        action,
+        RecordEntry::Hidden,
+    ));
+
+    // Double check that entry is not accessible in the RegisterAgentActivity op,
+    // which does get published.
+    assert!(matches!(
+        register_agent_activity_op.as_chain_op().unwrap().entry(),
+        RecordEntry::Hidden
+    ));
+
+    // Write all three ops to the database.
+    let register_agent_activity_op_hash = register_agent_activity_op.as_hash().clone();
+    let store_entry_op_hash = store_entry_op.as_hash().clone();
+    let store_record_op_hash = store_record_op.as_hash().clone();
+    vault
+        .write_async({
+            move |txn| -> StateMutationResult<()> {
+                holochain_state::mutations::insert_op_authored(txn, &register_agent_activity_op)?;
+                holochain_state::mutations::insert_op_authored(txn, &store_entry_op)?;
+                holochain_state::mutations::insert_op_authored(txn, &store_record_op)?;
+                Ok(())
+            }
+        })
+        .await
+        .unwrap();
+
+    // RegisterAgentActivity and StoreRecord are expected to be published.
+    // StoreRecord contains the entry and is expected to not be published.
+    let mut network = MockHolochainP2pDnaT::new();
+    let agent2 = agent.clone();
+    network.expect_publish().returning(
+        move |_request_validation_receipt,
+              _countersigning_session,
+              _basis_hash,
+              source,
+              op_hash_list,
+              _timeout_ms,
+              _reflect_ops| {
+            assert_eq!(source, agent2);
+            assert!(
+                op_hash_list.contains(&register_agent_activity_op_hash)
+                    || op_hash_list.contains(&store_record_op_hash)
+            );
+            assert!(!op_hash_list.contains(&store_entry_op_hash));
+            Ok(())
+        },
+    );
+    let network = Arc::new(network);
+
+    let (tx, _rx) =
+        TriggerSender::new_with_loop(Duration::from_secs(5)..Duration::from_secs(30), true);
+
+    let work_complete = publish_dht_ops_workflow(
+        vault.clone(),
+        network.clone(),
+        tx.clone(),
+        agent.clone(),
+        ConductorTuningParams::default().min_publish_interval(),
+    )
+    .await
+    .unwrap();
+
+    // Complete just means there have not been errors during publish.
+    assert_eq!(WorkComplete::Complete, work_complete);
```

Comment:
Shouldn't we be able to assert here that we sent somebody a record with the entry `::Hidden`?

Reply:
That's asserted [here](https://github.com/holochain/holochain/pull/4811/files#diff-e013207c1f6529187fb74c0b209ec6636ce35d019c89f2df149becaf0282be65R342-R346), or are you referring to something else?

The assertion could be rephrased to check that published ops always have `::Hidden` as the entry.

Reply:
Ahh, I see! I usually like to avoid assertions in a callback because it makes it hard to read the test as AAA or Given-When-Then but in this case it's checking an invariant so it makes sense. Looks good